### PR TITLE
Mention Structured Fields

### LINKED ISSUE #539: Semantics: mention SH syntax in "15.3.3. Considerations for New Field Values"?

### PR DESCRIPTION
**Pull request by mnot**
Including a re-org of Considerations for New Fields

Fixes #539.

### CODE REVIEW

File: draft-ietf-httpbis-semantics-latest.xml [hunk 1]
```diff
@@ -9746,54 +9806,36 @@ Content-Type: text/plain
...
+<t>
+   For example, the Content-Type field value only allows commas inside quoted
+   strings, and so can be reliably parsed, even when multiple values are
+   present. The Location field value provides a counter-example that should
+   not be emulated: because URIs can include commas, it is not possible to
+   reliably distinguish between a single value that includes a comma from two
+   values.
+<t>
+   Authors of fields with a singleton value (see <xref
+   target="field-values"/>) are additionally advised document how to treat
```
Comment (reschke) [inline draft-ietf-httpbis-semantics-latest.xml]: ...to document...

**Review by reschke — APPROVED** [review]

File: draft-ietf-httpbis-semantics-latest.xml [hunk 2]
```diff
@@ -9711,6 +9711,57 @@ Content-Type: text/plain
 </t>
 </section>
 
+<section title="Considerations for New Fields" anchor="considerations.for.new.fields">
+<t>
+   HTTP header and trailer fields are a widely-used extension point for the protocol.
+   While they can be (and often are) used in an ad hoc fashion, fields that are
+   widely used need to be more carefully documented, to ensure interoperability.
```
Comment (royfielding) [inline draft-ietf-httpbis-semantics-latest.xml:9718]:
```suggestion
   While they can be used in an ad hoc fashion, fields that are intended for
   wider use need to be carefully documented to ensure interoperability.
```

File: draft-ietf-httpbis-semantics-latest.xml [hunk 3]
```diff
@@ -9746,54 +9797,45 @@ Content-Type: text/plain
...
+   and HTTP libraries can perform combination automatically, this applies to
+   all field values &mdash; even when only a single value is anticipated.
+</t>
+<t>
+   Therefore, authors are advised to either delimit values that contain
+   commas (e.g., with the <x:ref>quoted-string</x:ref> rule; see <xref
+   target="quoted.strings"/>, or the String data type in <xref
+   target="RFCSTRF"/>), or encode or escape them. This ensures that commas in
+   the payload of the field value are not confused with those used to delimit
+   list values.
```
Comment (royfielding) [inline draft-ietf-httpbis-semantics-latest.xml:9822]:
```suggestion
   Therefore, authors are advised to delimit or encode values that contain
   commas (e.g., with the <x:ref>quoted-string</x:ref> rule of
   <xref target="quoted.strings"/>, the String data type of
   <xref target="RFCSTRF"/>), or a field-specific encoding).
   This ensures that commas within field data are not confused
   with the commas that delimit a list value.
```

File: draft-ietf-httpbis-semantics-latest.xml [hunk 4]
```diff
@@ -9746,54 +9797,45 @@ Content-Type: text/plain
...
+</t>
+<t>
+   For example, the <x:ref>Content-Type</x:ref> field value only allows commas
+   inside quoted strings, and so can be reliably parsed, even when multiple
```
Comment (royfielding) [inline draft-ietf-httpbis-semantics-latest.xml:9826]:
```suggestion
   inside quoted strings, which can be reliably parsed even when multiple
```

**Review by royfielding — CHANGES_REQUESTED** [review]

**Review by royfielding — APPROVED** [review]

**Pull request merged**